app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -240,6 +240,16 @@ class PatientSearchResultItemView(
     data class Plain(override val patientNumber: String) : PhoneNumber(patientNumber)
   }
 
+  sealed class Id(open val value: String) {
+
+    data class Highlighted(
+        override val value: String,
+        val highlightStart: Int,
+        val highlightEnd: Int
+    ) : Id(value)
+
+    data class Plain(override val value: String) : Id(value)
+  }
```

Comment:
Hmm, it seems like we have repeated this similar structure of having a "search criteria" and whether it's highlighted or not in many places.

Maybe a better solution for the future (not necessary to change in this PR) would be to use composition over inheritance.'

```kotlin
sealed class SearchResultItem {
  data class Name(): SearchResultItem()
  data class Phone(): SearchResultItem()
  data class Id(): SearchResultItem()
}

sealed class SearchResultItemType(val result: SearchResultItem) {
  data class Plain(): SearchResultItemType()
  data class Highlighted(): SearchResultItemType()
}
```

That way, whenever we need to implement a new type of search, we will need to implement a highlight by default as well.

What do you think?

Reply:
Yep, I was actually exploring the option of creating a specific type and simplifying this process. But that seemed like a different change and should be a different PR. So I will be creating a new change with those changes.